packages/hero/carousel.module.css

```diff
@@ -147,3 +147,15 @@
     }
   }
 }
+
+:global(.light) .root {
```

Comment:
Thanks for catching this - facepalming over here cause I think I just ripped out the `light` theme and forgot to add it  back in 🤦 .

I think it might be preferable to keep CSS grouped by selector (some rationale for this in our Notion doc under `Code Style` / `CSS Modules`), rather than grouped by variation. One way to do this would be to replace the global `"light" | "dark"` class with `s[backgroundTheme]` classes on affected elements. Another way might be to only pass that class to the root, and set CSS custom properties that can then be hooked into further down in the cascade. [Here's a commit](https://github.com/hashicorp/react-components/pull/295/commits/cad9e8fc76cd391022cd1262546b42d5ade5797f) `cad9e8f ` you could cherry pick for the former option, and [a preview of the result](https://react-components-storybook.vercel.app/?path=/story/components-hero--light-theme).

Reply:
Okay so I'm a little bit on the fence about this, and maybe I've missed the boat on the discussion 😅 .

Looking at the commit you referenced, that to me seems like we are erring too much on the side of composability and re-usability at the component level. I think having something like the `theme` prop at the level of an exportable component makes sense, but if we're drilling it all the way down into the child components that are not exported or really re-usable it seems like we're losing some of the benefits of the cascade.

I'll try out putting the class at the root and setting properties, that seems like a reasonable approach!

It seems like there could be room for a `ThemeProvider` of some sort that helps with this 💭

Reply:
Actually went with your commit for now, most in line with our current guidelines and consistency is a win 👍

Reply:
aack sorry i missed the comment earlier about the stripes being the reason for placement above - that's a great callout 💖 tbh not sure how to handle - looks like we have problems with the stripes on mobile anyways... but probably makes sense to prioritize desktop, so maybe we should add a `controlsFirst` prop or something, or maybe just always put them first?  🤔 augh i hate the number of options in this component ahah 😭

+1 on using the cascade by putting the class at the root and setting properties - or using a ThemeProvider if there are smaller components to drill into 💯 . I feel like i probably lean too heavily on just making things components sometimes, since I haven't used providers a ton yet (noting as context for why i went the direction i did and for future work ahah 😬 )

i probably also should have been more clear that's more just my hot take that i wrote down in Notion re: CSS, and not a discussion that's had a lot of activity, so super open to input and to changing our guidelines & recommendations there. I think your take of something more cascade-y for theming (whether through CSS modules or ThemeProvider of some sort) should be our first recommendation as a way to group styles by selector 👍  (also open to revising the recommendation to group styles by selector rather than state or theme, although that one does feel a little more solid IMO).